packages/pyright-internal/src/analyzer/service.ts

```diff
@@ -1374,9 +1382,16 @@ export class AnalyzerService {
         const envMarkers = [['bin', 'activate'], ['Scripts', 'activate'], ['pyvenv.cfg'], ['conda-meta']];
         const results: Uri[] = [];
         const startTime = Date.now();
-        const longOperationLimitInSec = 10;
-        const nFilesToSuggestSubfolder = 50;
-
+        // Use configurable timeout or default to 10 seconds
+        const longOperationLimitInSec =
+            this._configOptions.fileEnumerationTimeoutInSec !== undefined
+                ? this._configOptions.fileEnumerationTimeoutInSec
+                : 10;
+        // Use configurable min files or default to 50 files
+        const nFilesToSuggestSubfolder =
+            this._configOptions.fileEnumerationMinimumFiles !== undefined
+                ? this._configOptions.fileEnumerationMinimumFiles
+                : 50;
```

Comment:
nit: positive checks are a bit more readable than negative checks in if statements / ternaries imo
```suggestion
        // Use configurable timeout or default to 10 seconds
        const longOperationLimitInSec =
            this._configOptions.fileEnumerationTimeoutInSec === undefined
                ? 10
                : this._configOptions.fileEnumerationTimeoutInSec;
        // Use configurable min files or default to 50 files
        const nFilesToSuggestSubfolder =
            this._configOptions.fileEnumerationMinimumFiles === undefined
                ? 50
                : this._configOptions.fileEnumerationMinimumFiles;
```